src/Modal/index.js

```diff
@@ -0,0 +1,266 @@
+   // =======================================================
+   // Styles for Modal Window
+   // =======================================================
+
+   const windowProps = {
+      width,
+      height,
+      backgroundColor,
+      borderRadius,
+      boxShadow,
+      margin,
+      maxHeight,
+      maxWidth,
+      minHeight,
+      minWidth
+   }
+
+   // =======================================================
+   // Open and Close Events
+   // =======================================================
+
+   // When open animation has completed
+   const handleOpenComplete = () => {
+      if (onOpenComplete) {
+         onOpenComplete()
+      }
+   }
+
+   // When Close Animation has completed
+   const handleCloseComplete = () => {
+      modalIsActive.current = false
+      setIsClosing(false)
+      portal.removeAttribute('style')
+
+      if (onCloseComplete) {
+         onCloseComplete()
+      }
+   }
+
+   // Currently this is only triggered by clicking on overlay
+   // TODO: Handle if user presses escape key
+   const handleCloseTrigger = () => {
+      if (onCloseTrigger) {
+         onCloseTrigger()
+      }
+   }
+
+   // =======================================================
+   // Handle changes to open prop
+   // =======================================================
+
+   useEffect(() => {
+      if (open) {
+         // We need to ensure the portal container has the following style
+         // to ensure that the modal displays correctly and fills the entire screen
+         // Note: we ONLY want it to have this style when modal is open,
+         // otherwise it may cover content  while modal is closed
+         portal.setAttribute(
+            'style',
+            'position: fixed; top: 0; left: 0; right: 0; bottom: 0;'
+         )
+
+         if (onOpen) {
+            onOpen()
+         }
+
+      // If Modal open changes from true to false,
+      // begin close animation only if Modal was previously open
+      } else if (modalIsActive.current) {
+         setIsClosing(true)
+
+         if (onClose) {
+            onClose()
+         }
+      }
+
+      return () => {
+         portal.removeAttribute('style')
+      }
+   }, [open])
+
+   // Cleanup on unmount
+   useEffect(() => () => portal.remove(), [])
+
+   // =======================================================
+   // Render
+   // =======================================================
+
+   return ReactDOM.createPortal(
+      <View
+         display={modalIsActive.current ? 'flex' : 'none'}
+         css={{ pointerEvents: shouldPropagatePointerEvents ? 'none' : 'auto' }}
+         justifyContent="center"
+         alignItems="center"
+         position="absolute"
+         top="0px"
+         left="0px"
+         right="0px"
+         bottom="0px"
+         zIndex={zIndex}>
+
+         {/* When Modal is Open, we need to prevent body from scrolling */}
+         {modalIsActive.current && <Global styles={{ body: { overflow: 'hidden' } }} />}
```

Comment:
Hmm this seems a little overkilley, no? Because to the best of my knowledge, emotion will add (and remove) a `style` tag to the dom each time this is toggled. Might be more lightweight to do:

```js
useEffect(() => {
   if (modalIsActive.current) {
      document.body.style.overflow = 'hidden'
   } else {
      document.body.style.overflow = undefined or 'auto' or whatever
   }
}, [modalIsActive.current])
```